packages/office-ui-fabric-react/src/components/List/List.tsx

```diff
@@ -333,31 +342,48 @@ export class List<T = any> extends React.Component<IListProps<T>, IListState<T>>
     }
   }
 
+  public componentDidUpdate(): void {
+    // Multiple updates may have been queued, so the callback will reflect all of them.
+    // Re-fetch the current props and states to avoid using a stale props or state captured in the closure.
+    const finalProps = this.props;
```

Comment:
unclear why you need to variable-ize these. Is some magic getter logic happening? Not sure why you can't just destructure the values you need here:

```
const { pages } = this.state;
```

Reply:
As part of porting over the code, I did my best to remove all direct calls to `this.props` and `this.state` so I could keep track of them. This is because many function now need to take those values purely as function parameters because `this.props` could be the wrong set of values. What you see here is literally what was in the `setState` commit callback, when this code was in `componentWillReceiveProps`.